src/components/autocompleter/autocompleter.test.jsx

```diff
@@ -0,0 +1,139 @@
+      expect(screen.getByLabelText('Bar')).toBeDefined();
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('accepts a placeholder', () => {
+      render(<Autocompleter {...requiredProps} placeholder="I am place" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('placeholder', 'I am place');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('readonly', '');
+    });
+
+    it('unsets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly={false} />);
+      expect(screen.getByLabelText('Test label')).not.toHaveAttribute('readonly', '');
+    });
+  });
+
+  describe('required API', () => {
+    it('sets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required />);
+      expect(screen.getByLabelText('Test label')).toBeRequired();
+    });
+
+    it('unsets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required={false} />);
+      expect(screen.getByLabelText('Test label')).not.toBeRequired();
+    });
+  });
+
+  describe('value API', () => {
+    it('accepts a value', async () => {
+      const value = { id: 22, name: 'cool dude' };
+      render(<Autocompleter {...requiredProps} value={value} />);
+
+      expect(screen.getByLabelText('Test label')).toHaveValue('cool dude');
+    });
+  });
+
+  describe('models API', () => {
+    it('accepts a an array of models', async () => {
+      const models = [{ id: 22, name: 'cool dude' }, { id: 33, name: 'neato burrito' }];
+      render(<Autocompleter {...requiredProps} models={models} />);
+
+      userEvent.click(screen.getByLabelText('Test label'));
+      expect(screen.queryByText('cool dude')).toBeInTheDocument();
+      expect(screen.queryByText('neato burrito')).toBeInTheDocument();
+    });
+  });
+
+  describe('forwardRef API', () => {
+    it('can be used to get value as array of selected id', async () => {
+      const ref = createRef();
+      render(<Autocompleter {...requiredProps} value={{ name: 'neat', id: 11 }} ref={ref} />);
+
+      expect(ref.current.value).toStrictEqual({ name: 'neat', id: 11 });
+    });
+  });
+
+  describe('self loading models', () => {
+    it('fetches the models on mount', async () => {
+      render(<Autocompleter {...requiredProps} />);
+
+      userEvent.click(screen.getByLabelText('Test label'));
+
+      expect(await screen.findByText('Fizz')).toBeInTheDocument();
+      expect(screen.getByText('Buzz')).toBeInTheDocument();
+    });
+  });
+
+  describe('search', () => {
+    it('searches with the search string', async () => {
+      render(<Autocompleter {...requiredProps} />);
+
+      userEvent.click(screen.getByLabelText('Test label'));
+      userEvent.type(document.activeElement, 'Ba');
```

Comment:
I wonder if there is room (in this repo or maybe in user-event public repo) to type or interact with the focused element by default.

I always find it odd that for `type` and `blur` and some other events I have to pass the element in, I would expect the focused element to be the default.

Reply:
You just have to go lower level: https://github.com/testing-library/user-event#keyboardtext-options

Reply:
I think this is just one of those things that becomes strange when you're writing an API to mimic user interaction, but for automated and code-driven tests. Personally, I like the explicit-ness of knowing what I'm "typing" into.